File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/Manipulator.java
package org.firstinspires.ftc.teamcode.robot;

import com.qualcomm.robotcore.hardware.Servo;

public class Manipulator {
    Robot robot;

    public Servo ManipulatorServoLeft;
    public Servo ManipulatorServoRight;
    public Servo VerticalServo;
    public Servo Samolet;
    public Servo LeftHangServo;
    public Servo RightHangServo;

    public double LeftmaxPos = 0.5;
    public double LeftminPos = 0.2;
    public double RightmaxPos = 0.5;
    public double RightminPos = 0.2;
    public double increment = 0.1;
    public double openPos = 0.1;

    public Manipulator(Robot robot) {
        this.robot = robot;

        ManipulatorServoLeft = robot.linearOpMode.hardwareMap.get(Servo.class, "left_manipulator_servo");
        ManipulatorServoRight = robot.linearOpMode.hardwareMap.get(Servo.class, "right_manipulator_servo");
        VerticalServo = robot.linearOpMode.hardwareMap.get(Servo.class, "vertical_servo");
        Samolet = robot.linearOpMode.hardwareMap.get(Servo.class, "samolet");
        LeftHangServo = robot.linearOpMode.hardwareMap.get(Servo.class, "left_hang_servo");
        RightHangServo = robot.linearOpMode.hardwareMap.get(Servo.class, "right_hang_servo");

        ManipulatorServoLeft.setDirection(Servo.Direction.FORWARD);
        ManipulatorServoRight.setDirection(Servo.Direction.REVERSE);
    }

    public void setPosAll(double posLeft, double posRight) {
        ManipulatorServoLeft.setPosition(posLeft);
        ManipulatorServoRight.setPosition(posRight);
    }

    public void setPosLeft(double pos) {
        ManipulatorServoLeft.setPosition(pos);
    }

    public void setPosRight(double pos) {
        ManipulatorServoRight.setPosition(pos);
    }

    public void setVerticalPos(double pos) {
        VerticalServo.setPosition(pos);
    }

    public void moveSamolet(double pos) {
        Samolet.setPosition(pos);
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/Manipulator.kt
package org.firstinspires.ftc.teamcode.robot

import com.qualcomm.robotcore.hardware.Servo

class Manipulator(var robot: Robot) {
    @JvmField
    var ManipulatorServoLeft: Servo = robot.linearOpMode.hardwareMap.get(Servo::class.java, "left_manipulator_servo")
    @JvmField
    var ManipulatorServoRight: Servo = robot.linearOpMode.hardwareMap.get(Servo::class.java, "right_manipulator_servo")
    @JvmField
    var MoveServo: Servo = robot.linearOpMode.hardwareMap.get(Servo::class.java, "move_servo")
    @JvmField
    var Samolet: Servo = robot.linearOpMode.hardwareMap.get(Servo::class.java, "samolet")
    @JvmField
    var LeftHangServo: Servo = robot.linearOpMode.hardwareMap.get(Servo::class.java, "left_hang_servo")
    @JvmField
    var RightHangServo: Servo = robot.linearOpMode.hardwareMap.get(Servo::class.java, "right_hang_servo")

    @JvmField
    val LeftmaxPos = 0.5
    @JvmField
    val LeftminPos = 0.2
    @JvmField
    val RightmaxPos = 0.5
    @JvmField
    val RightminPos = 0.2
    @JvmField
    val increment = 0.1;
    @JvmField
    val openPos = 0.1;




    fun setPosAll(posl: Double, posr: Double) {
        ManipulatorServoLeft.direction = Servo.Direction.FORWARD
        ManipulatorServoRight.direction = Servo.Direction.REVERSE
        ManipulatorServoLeft.position = posl
        ManipulatorServoRight.position = posr
    }
    fun setPos(pos: Double) {
        ManipulatorServoLeft.position = pos
        ManipulatorServoRight.position = pos
    }
    fun moveSetPos(pos: Double) {
        MoveServo.position = pos
    }
    fun moveSamolet(pos: Double) {
        Samolet.position = pos
    }
}